src/main/java/mil/dds/anet/search/mssql/MssqlAuthorizationGroupSearcher.java

```diff
@@ -0,0 +1,97 @@
+import mil.dds.anet.beans.search.ISearchQuery.SortOrder;
+import mil.dds.anet.beans.search.AuthorizationGroupSearchQuery;
+import mil.dds.anet.beans.search.AuthorizationGroupSearchQuery.AuthorizationGroupSearchSortBy;
+import mil.dds.anet.database.mappers.AuthorizationGroupMapper;
+import mil.dds.anet.search.IAuthorizationGroupSearcher;
+import mil.dds.anet.utils.DaoUtils;
+import mil.dds.anet.utils.Utils;
+
+public class MssqlAuthorizationGroupSearcher implements IAuthorizationGroupSearcher {
+
+	@Override
+	public AuthorizationGroupList runSearch(AuthorizationGroupSearchQuery query, Handle dbHandle) {
+		final List<String> whereClauses = new LinkedList<String>();
+		final Map<String,Object> sqlArgs = new HashMap<String,Object>();
+		final StringBuilder sql = new StringBuilder("/* MssqlAuthorizationGroupSearch */ SELECT authorizationGroups.*");
+
+		final String text = query.getText();
+		final boolean doFullTextSearch = (text != null && !text.trim().isEmpty());
+		if (doFullTextSearch) {
+			// If we're doing a full-text search, add a pseudo-rank (the sum of all search ranks)
+			// so we can sort on it (show the most relevant hits at the top).
+			// Note that summing up independent ranks is not ideal, but it's the best we can do now.
+			// See https://docs.microsoft.com/en-us/sql/relational-databases/search/limit-search-results-with-rank
+			sql.append(", ISNULL(c_authorizationGroups.rank, 0) + ISNULL(f_authorizationGroups.rank, 0)");
+			sql.append(" AS search_rank");
+		}
+		sql.append(", count(*) over() as totalCount FROM authorizationGroups");
+
+		if (doFullTextSearch) {
+			sql.append(" LEFT JOIN CONTAINSTABLE (authorizationGroups, (name, description), :containsQuery) c_authorizationGroups"
+					+ " ON authorizationGroups.id = c_authorizationGroups.[Key]"
+					+ " LEFT JOIN FREETEXTTABLE(authorizationGroups, (name, description), :freetextQuery) f_authorizationGroups"
+					+ " ON authorizationGroups.id = f_authorizationGroups.[Key]");
+			whereClauses.add("c_authorizationGroups.rank IS NOT NULL");
+			sqlArgs.put("containsQuery", Utils.getSqlServerFullTextQuery(text));
+			sqlArgs.put("freetextQuery", text);
+		}
+
+		if (query.getStatus() != null) {
+			whereClauses.add("authorizationGroups.status = :status");
+			sqlArgs.put("status", DaoUtils.getEnumId(query.getStatus()));
+		}
+
+		final AuthorizationGroupList result = new AuthorizationGroupList();
+		result.setPageNum(query.getPageNum());
+		result.setPageSize(query.getPageSize());
+
+		if (whereClauses.isEmpty()) {
+			return result;
+		}
+
+		sql.append(" WHERE ");
+		sql.append(Joiner.on(" AND ").join(whereClauses));
+
+		//Sort Ordering
+		final List<String> orderByClauses = new LinkedList<>();
+		if (doFullTextSearch && query.getSortBy() == null) {
+			// We're doing a full-text search without an explicit sort order,
+			// so sort first on the search pseudo-rank.
+			orderByClauses.addAll(Utils.addOrderBy(SortOrder.DESC, null, "search_rank"));
```

Comment:
@gjvoosten This is a dependency on the Fast Search commit which has been reverted (for now). Needs updating